compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/Scrollable.kt

```diff
@@ -278,33 +285,157 @@ private fun Modifier.pointerScrollable(
         },
         canDrag = { down -> down.type != PointerType.Mouse }
     )
-        .mouseWheelScroll(scrollLogic, scrollConfig)
+        // TODO Make animation configurable
+        .animatedMouseWheelScroll(scrollLogic, scrollConfig)
         .nestedScroll(nestedScrollConnection, nestedScrollDispatcher.value)
 }
 
-private fun Modifier.mouseWheelScroll(
-    scrollingLogicState: State<ScrollingLogic>,
+@Composable
+private fun Modifier.rawMouseWheelScroll(
+    scrollLogic: State<ScrollingLogic>,
+    mouseWheelScrollConfig: ScrollConfig,
+) = mouseWheelInput(mouseWheelScrollConfig) {
+    scrollLogic.value.dispatchRawDelta(it) != Offset.Zero
+}
+
+@Composable
+private fun Modifier.animatedMouseWheelScroll(
+    scrollLogic: State<ScrollingLogic>,
     mouseWheelScrollConfig: ScrollConfig,
-) = pointerInput(scrollingLogicState, mouseWheelScrollConfig) {
-    awaitPointerEventScope {
```

Comment:
Additional not about this code.

When we have
```
awaitPointerEventScope {
  while (true) {
     awaitPointerEvent()
  }
}
```
we don't miss any scroll event. It means, if there were 20 scroll events during one frame, we recieve them all. But because it was slow to handle each event in the lazy list (at least in 1.1), we had to handle only the first received event and drop the reset this way:
```
while (true) {
   awaitPointerEventScope {
     awaitPointerEvent()
  }
}
```

It seems it was reverted in the Android after that: https://android-review.googlesource.com/c/platform/frameworks/support/+/2087863, because they had more issues (I only noticed this now).

Ideally, we shouldn't drop events, but batch them into one, and handle the total delta. 

In your fix, you again drop events:
```
while (isActive) {
  val event = awaitPointerEventScope {
     awaitScrollEvent()
  }
```
And it is probably good for us now, but we should think about it more during upstreaming to Jetpack Compose for Android (that we will do someday).